app/models/filters/filter_base.rb

```diff
@@ -573,10 +589,25 @@ def effective_project_ids
         # Add an invalid id if there are none
         project_ids = [0] if project_ids.empty?
 
-        project_ids.uniq.reject(&:blank?)
+        reject_excluded_project_ids(project_ids.uniq.reject(&:blank?))
       end
     end
 
+    def reject_excluded_project_ids(project_ids)
+      return project_ids if excluded_project_ids.empty? && excluded_project_type_numbers.empty?
+
+      project_ids.reject! { |id| excluded_project_ids.include?(id) }
```

Comment:
probably better not to mutate our inputs, could cause weird bugs
```suggestion
      project_ids = project_ids.reject { |id| excluded_project_ids.include?(id) }
```